apis/addonconfigs/cni/v1alpha2/antreaconfig_types.go

```diff
@@ -288,51 +288,169 @@ type AntreaNsx struct {
 	// +kubebuilder:validation:Optional
 	// +kubebuilder:default:=false
 	Enable bool `json:"enable,omitempty"`
-	// BootstrapFrom either providerRef or inline configs
-	// +kubebuilder:validation:Optional
-	BootstrapFrom AntreaNsxBootstrapFrom `json:"bootstrapFrom,omitempty"`
 	// Config is  configuration for nsxt adapter
 	// +kubebuilder:validation:Optional
 	AntreaNsxConfig AntreaNsxConfig `json:"config,omitempty"`
 }
 
-type AntreaNsxBootstrapFrom struct {
-	// ProviderRef is used with uTKG, which will be filled by uTKG Addon Controller
+type AntreaNsxConfig struct {
+	// BootstrapFrom either providerRef or inline configs
 	// +kubebuilder:validation:Optional
-	ProviderRef *AntreaNsxProvider `json:"providerRef,omitempty"`
-	// Inline is used with TKGm, user need to fill in manually
+	BootstrapFrom string `json:"bootstrapFrom,omitempty"`
+
 	// +kubebuilder:validation:Optional
-	Inline *AntreaNsxInline `json:"inline,omitempty"`
-}
+	BootstrapSupervisorResourceName string `json:"bootstrapSupervisorResourceName,omitempty"`
```

Comment:
I think we don't need to expose BootstrapSupervisorResourceName to CRD. BootstrapSupervisorResourceName is created by the antrea addon controller itself.